common/txmgr/address_state.go

```diff
@@ -247,12 +247,51 @@ func (as *addressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) findTx
 func (as *addressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) pruneUnstartedTxQueue(ids []int64) {
 }
 
+// reapConfirmedTxs removes confirmed transactions that are older than the given time threshold.
+// It also removes confirmed transactions that are older than the given block number threshold.
```

Comment:
Feel like the comment makes it sound like the tx reaped are either older than the time threshold or older than the block number. Really it should convey that both criteria have to be met
```suggestion
// reapConfirmedTxs removes confirmed transactions that are older than the given time threshold and have receipts older than the given block number threshold.
```